manim/utils/ipython_magic.py

```diff
@@ -0,0 +1,61 @@
+"""Utilities for using Manim with IPython (in particular: Jupyter notebooks)"""
+
+import mimetypes
+from pathlib import Path
+
+from IPython.core.magic import Magics, magics_class, line_cell_magic, needs_local_scope
+from IPython.display import HTML, Image
+
+from manim import config, tempconfig
+
+from .._config.main_utils import parse_args
+
+
+@magics_class
+class ManimMagic(Magics):
+    @needs_local_scope
+    @line_cell_magic
+    def manim(self, line, cell=None, local_ns=None):
+        r"""Render Manim scenes contained in Jupyter cells.
+        Works as a line or cell magic.
+
+        Usage, in line mode::
+
+            %manim MyAwesomeScene [CLI options]
+
+        Usage, in cell mode::
+
+            %%manim MyAwesomeScene [CLI options]
+
+            class MyAweseomeScene(Scene):
+                def construct(self):
+                    ...
+
+        Run ``%manim -h`` for possible command line interface options.
```

Comment:
Regarding the place for the docsting:
What about
https://docs.manim.community/en/latest/reference.html#other-modules 
?
Further, it reminds me that we were also once searching for the right place for manim_directive:
https://docs.manim.community/en/latest/reference/manim_directive.html
So, maybe one could put them somewhere together in a new category?

Reply:
The fundamental difference between `manim_directive` and this new module is that the IPython magic module needs to be included in the main library, otherwise we won't be able to ship and package it with the other library code.

Thus, when it comes to the reference manual, it should be listed where it lives (which currently is under `manim.utils`). What I meant above was not where in the reference manual the documentation for the module should be included, but rather where else (e.g., tutorials?) we should mention that we support running Manim from within Jupyter notebooks.

Reply:
Ah, okay, now I get it.
The tutorials would be a good place, and I think it should be also mentioned in the readme.md
Further, we can make a post on discord/twitter on this jupyter support.

Reply:
I've added some documentation in the latest commit. I'll make sure that it builds and that the tests pass again, but then I am done with this PR.